src/core/component.js

```diff
@@ -200,20 +200,27 @@ Component.prototype = {
    * @param {string} attrValue - HTML attribute value.
    *        If undefined, use the cached attribute value and continue updating properties.
    */
-  updateProperties: function (attrValue) {
+  updateProperties: function (attrValue, clobber) {
     var el = this.el;
     var isSinglePropSchema = isSingleProp(this.schema);
     var oldData = extendProperties({}, this.data, isSinglePropSchema);
+    var skipTypeChecking = typeof this.previousAttrValue === 'object' && attrValue === this.previousAttrValue;
+    var currentAttrValue = !clobber && this.attrValue;
+    this.previousAttrValue = attrValue;
 
-    if (attrValue !== undefined) { this.updateCachedAttrValue(attrValue); }
+    if (!el.hasLoaded) {
+      if (attrValue !== undefined) { this.updateCachedAttrValue(attrValue); }
+      return;
+    }
 
-    if (!el.hasLoaded) { return; }
+    attrValue = this.parseAttrValueForCache(attrValue);
 
     if (this.updateSchema) {
-      this.updateSchema(buildData(el, this.name, this.attrName, this.schema, this.attrValue,
-                                  true));
+      this.updateSchema(this.buildData(attrValue, currentAttrValue, true));
     }
-    this.data = buildData(el, this.name, this.attrName, this.schema, this.attrValue);
+    this.data = this.buildData(attrValue, currentAttrValue, false, skipTypeChecking);
```

Comment:
Not concerned about the number of flags, but being explicit, because calling a function `build(value, oldValue, true, false)`, hard to know what `true`/`false` does in that order.

But maybe not so we don't create another object.